### DISCUSSION TITLE
Invalid Route error when triggering request model action after adding new edge

### DISCUSSION THREAD

**Discussion by jmenzies12** [eclipse-glsp/glsp#1420] [Q&A] [answered]
Hey there,

I have hooked up GLSP diagrams to a Langium source model. My ModelStorage triggers a RequestModelAction when it receives a document change event from the Langium language client. When I create a new edge or node, I send an event out to Langium to create it and then it triggers a document change, this is picked up by my ModelStorage which then recreates my diagram model from the updated source model.

For the most part this works okay, however I get an invalid route error from the layout utils via a ChangeBoundsAction whenever I create a new edge in the diagram. I'm assuming doing it this way is bypassing some code that adds the routing points but I can't figure out what or how to resolve it. I tried adding routing points in my gmodel factory to ensure that they were always added but that had no effect because the newRoutingPoints on the ChangeBoundsAction was still empty. I also tried adding a placeholder edge to my diagram model, which does work in displaying the new edge but I still get the invalid route error.

Any thoughts? 

Kind Regards
Jonathan

**Comment by martin-fleck-at** [comment]
Hi Jonathan,

There is a very similar setup in [CrossModel](https://github.com/CrossBreezeNL/crossmodel/tree/main). However, I cannot see any particular handling of the routing points in the [creation of a relationship](https://github.com/CrossBreezeNL/crossmodel/blob/main/extensions/crossmodel-lang/src/glsp-server/system-diagram/handler/create-relationship-operation-handler.ts) or in the change bounds action. Typically, routing points are created using the [calcElementAndRoutingPoints](https://github.com/eclipse-glsp/glsp-client/blob/master/packages/client/src/utils/gmodel-util.ts#L309) utility function which in turn uses registered edge routers as part of the standard [routingModule](https://github.com/eclipse-glsp/glsp-client/blob/69538359964d5572654c1c68160f43bd48f11674/packages/client/src/features/routing/routing-module.ts#L36C14-L36C27). This is what gets send to the server. Do you set a particular `routerKind` on your edges? Which version of GLSP are you using and what is the concrete error?

Kind regards,
Martin

**Reply by jmenzies12** [reply, marked as answer]
Hi Martin,

Thanks for the response, oddly after fixing some issues with my nodes I haven't ran into this problem again.

Kind Regards,
Jonathan
